.github/workflows/global-cspell.yml

```diff
@@ -0,0 +1,27 @@
+name: global-cspell
```

Comment:
Do we want to have a global cspell?
It looks like I added this `cspell` config as part of https://github.com/Azure/azure-dev/pull/74/files but the intention is to lint the /templates folder only. I might've forgot to add the `--root` argument to limit this be templates only (See the name of the step on line 23 `Run spell check for templates`).

So, we are mixing a little bit the concept of the `global cspell settings` with the cspell steps for each project in the repo.
The global config is meant to apply to all projects, as it is referenced from each project's cspell config.

I think that the right approach is to keep adding a `.vscode` folder within each project and a cspell unique config for that project that reference the global config.

Then, we need to push the cspell-lint step to its own file (like we do with `lint-bicep`) and re-use it from all the flows that runs cspell, just by setting the regex for the included files, the config file and the root folder

Reply:
I'll let @danieljurek comment on this.

It just depends on whether we treat `templates` as a separate project. My view is that `templates` isn't self-contained. A template can reference via `repoman` files from outside of the `templates` folder, thus is it possible to lint "templates" alone?

I do agree with using the project-specific layout for new projects, but I have a hard time viewing `templates` as a standalone project with a `.vscode` folder that people would choose to operate in.

Even if we did separate templates into its own cspell. We still need a global "catch-all" for the remaining files that are outside of templates. So as a pragmatic approach, it's combined into one since the runtime cost is a few seconds for the combined approach.

Reply:
Spell checking the entire repository takes ~24 seconds. It's pretty cheap. One cspell workflow would prevent us from having to make a bunch of workflows which each has to be updated as things change. 

Can we get a cspell config which, when run from the root of the repo, scans all files in the repo and applies rules defined in domain-specific files? (i.e. golang dictionary for `cli/azd`, JS dictionary for the react app, etc.)